src/outputs/parthenon_hdf5.cpp

```diff
@@ -841,49 +844,48 @@ void PHDF5Output::WriteOutputFileImpl(Mesh *pm, ParameterInput *pin, SimTime *tm
     const hsize_t nx5 = vinfo.nx5;
     const hsize_t nx4 = vinfo.nx4;
 
-    local_count[4] = global_count[4] = vlen;
+    local_count[1] = global_count[1] = vlen;
 
     // load up data
     hsize_t index = 0;
 
     // for each local mesh block
     for (size_t b_idx = 0; b_idx < num_blocks_local; ++b_idx) {
       const auto &pmb = pm->block_list[b_idx];
-      bool found = false;
+      bool is_allocated = false;
 
       // for each variable that this local meshblock actually has
       const auto vars = get_vars(pmb);
       for (auto &v : vars) {
-        // Note index l transposed to interior
         // For reference, if we update the logic here, there's also
         // a similar block in parthenon_manager.cpp
         if (v->IsAllocated() && (var_name == v->label())) {
           auto v_h = v->data.GetHostMirrorAndCopy();
-          for (int k = out_kb.s; k <= out_kb.e; ++k) {
-            for (int j = out_jb.s; j <= out_jb.e; ++j) {
-              for (int i = out_ib.s; i <= out_ib.e; ++i) {
-                for (int l = 0; l < nx6; ++l) {
-                  for (int m = 0; m < nx5; ++m) {
-                    for (int n = 0; n < nx4; ++n) {
-                      tmpData[index++] = static_cast<OutT>(v_h(l, m, n, k, j, i));
+          for (int t = 0; t < nx6; ++t) {
+            for (int u = 0; u < nx5; ++u) {
+              for (int v = 0; v < nx4; ++v) {
+                for (int k = out_kb.s; k <= out_kb.e; ++k) {
+                  for (int j = out_jb.s; j <= out_jb.e; ++j) {
+                    for (int i = out_ib.s; i <= out_ib.e; ++i) {
```

Comment:
Can we make this `nx3`, `nx2`, `nx1` to support variables that are not the same shape as a cell-centered var on the meshblock? I.e., that way we can automatically support faces, nodes, etc? I know that's not the point of this PR, but I think it's a small enough change we should just make it "while we're here."

Reply:
Given that I originally thought this PR was also "just a small change" but in the end there were quite a few not that obvious places, I decided to keep this separate and easier to review (especially as I also want to introduce a test case for the non cell centered output).
If you think it's better to combine I can also add the change here. Otherwise, I'll open a separate PR soonish (which should not affect downstream codes in terms of consistency as it just adds/fixes things).

Reply:
Ah ok. Two small PRs works for me. You're right that's probably the better way to go. Just wanted to make sure that the meshblock size thing didn't get lost in the noise.